### PR TITLE
Application frameworking

### PR DESCRIPTION
**Pull request by benmelz**
Sets up all the frameworking necessary for development, including:
- documentation (license/readme)
- dependencies (node-verison/npm)
- linters (eslint, stylelint)
- build tools (vite, PWA plugin)
- arcgis framework (basic webmap with basic widgets)
- ci/cd (travis lint, build and staging steps, dependabot)

### CODE REVIEW

File: public/robots.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+User-agent: *
+Disallow:
```
Comment (werebus) [inline public/robots.txt:2]: I'm not 100% sure that a blank `Disallow:` is correct. Perhaps should be `Allow: /`

File: .env [hunk 2]
```diff
@@ -0,0 +1,5 @@
+VITE_APP_NAME="UMass Amherst Parking Map"
+VITE_APP_SHORT_NAME="Parking Map"
+VITE_APP_DESCRIPTION="Interactive parking map of the University of Massachusetts Amherst campus."
```
Comment (werebus) [inline .env]: "for" rather that "of"? Also the period is bugging me since it's not a complete sentence.

File: .env [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+VITE_APP_BACKGROUND_COLOR="#ffffff"
+VITE_APP_THEME_COLOR="#ffffff"
```
Comment (werebus) [inline .env:5]: Could also be Maroon.

**Review by werebus — CHANGES_REQUESTED** [review]
Vite does say, "boy, that's a big JS bundle!" when building. But I don't know that there's anything to be done about it. Seems to be entirely up to Esri how much stuff is piled into `Map` and `MapView`.

File: public/robots.txt (hunk 1 above)
Comment (benmelz) [reply public/robots.txt:2]: I'm having trouble finding a nice authoritative source on this (probably would have to look into docs for each individual crawler) but some light searching suggests that at best `Disallow:` is more "common" and at worst `Allow: /` might not be understood by some?

https://www.conductor.com/academy/robotstxt/#all-robots-can-access-everything

https://stackoverflow.com/questions/70573337/what-is-the-difference-between-allow-disallow-in-robots-txt#:~:text=Disallow%20is%20part%20of%20the,txt.&text=In%20that%20case%2C%20most%20bots,to%20crawl%20the%20public%20directory.

https://stackoverflow.com/questions/4276957/how-to-configure-robots-txt-to-allow-everything (comments and second answer)

File: .env (hunk 3 above)
Comment (benmelz) [reply .env:5]: I plan on eventually doing this, but am waiting until some UMass branding is actually added (i.e. the banner/navbar).

**Comment by benmelz** [comment]
> Vite does say, "boy, that's a big JS bundle!" when building. But I don't know that there's anything to be done about it. Seems to be entirely up to Esri how much stuff is piled into `Map` and `MapView`.

Indeed, in my "roadmap" I'm thinking that I might try to tinker with the rollup chunking settings to try to address this around MVP time when we're more "stable"/have a more complete picture of the final build.

File: public/robots.txt (hunk 1 above)
Comment (werebus) [reply public/robots.txt:2]: The original (1994!) "specification" didn't _have_ `Allow:`. Turns out the [RFC](https://datatracker.ietf.org/doc/html/rfc9309) was published in 2022, a scant 28 years later. And that confirms that an empty `Disallow:` is fine. Although it also says "If an "allow" rule and a "disallow" rule are equivalent, then the "allow" rule SHOULD be used."

Paging through that RFC, all of these _should_ be equivalent:

```
User-agent: *
Allow: /
```

```
User-agent: *
Disallow:
```

```
User-agent: *
```

```
```

(No file at all)

All that to say, this is fine, and _might_ be the best choice for very, very, very old crawlers.

**Review by werebus — APPROVED** [review]

**Pull request merged**